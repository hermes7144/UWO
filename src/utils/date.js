// import { format, register } from 'timeago.js';
// import koLocale from 'timeago.js/lib/lang/ko';

// register('ko', koLocale);

// export function formatAgo(date, lang = 'en_US') {
//   return format(date, lang);
// }

export function formatDate(rawDate) {
  const date = new Date(rawDate);
  const year = date.getFullYear().toString().slice(-2); // 연도에서 뒤의 두 자리만 추출
  const month = ('0' + (date.getMonth() + 1)).slice(-2); // 월을 두 자리 숫자로 변환
  const day = ('0' + date.getDate()).slice(-2); // 일을 두 자리 숫자로 변환
  return `${year}/${month}/${day}`;
}