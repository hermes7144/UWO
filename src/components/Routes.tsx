import { useNavigate } from 'react-router-dom';
import { useRouteHooksContext } from '../context/RouteHooksContext';
import { formatDate } from '../utils/date';

type RouteType = {
  id: string;
  title: string;
  description: string;
  createdAt: number;
  citys: number[];
  country: string;
  region: string;
  startMonth: number;
  endMonth: number;
};

export default function Routes() {
  const { useRoute } = useRouteHooksContext();
  const navigate = useNavigate();

  const {
    routesQuery: { isLoading, error, data: routes },
  } = useRoute();

  return (
    <>
      {isLoading && <p>Loading...</p>}
      {error && <p>Something is wrong</p>}
      {routes && (
        <table className='min-w-full text-center text-sm font-light'>
          <colgroup>
            <col width={'5%'} />
            <col width={'5%'} />
            <col width={'5%'} />
            <col width={'50%'} />
            <col width={'10%'} />
            <col width={'10%'} />
          </colgroup>
          <thead className='border-b bg-neutral-50 font-medium dark:border-neutral-500 dark:text-neutral-800'>
            <tr>
              <th scope='col' className='px-1 py-4'>
                국가
              </th>
              <th scope='col' className='px-1 py-4'>
                지역
              </th>
              <th scope='col' className='px-1 py-4'>
                기간
              </th>
              <th scope='col' className='px-1 py-4'>
                제목
              </th>
              <th scope='col' className='px-1 py-4'>
                작성일
              </th>
              <th scope='col' className='px-1 py-4'>
                추천
              </th>
            </tr>
          </thead>
          <tbody>
            {routes.map((route: RouteType) => (
              <tr key={route.id} className='border-b dark:border-neutral-500'>
                <td className='whitespace-nowrap px-1 py-2'>{route.country}</td>
                <td className='whitespace-nowrap px-1 py-2'>{route.region}</td>
                <td className='whitespace-nowrap px-1 py-2'>
                  {route.startMonth} ~ {route.endMonth}월
                </td>
                <td className=' overflow-hidden px-2 py-4 text-left text-ellipsis whitespace-nowrap' style={{ maxWidth: '250px' }}>
                  <span
                    className='cursor-pointer hover:underline'
                    onClick={() => {
                      navigate(`/routes/${route.id}`);
                    }}>
                    {route.title}
                  </span>
                </td>
                <td className='whitespace-nowrap px-2 py-2'>{formatDate(route.createdAt)}</td>
                <td className='whitespace-nowrap  px-2 py-2'>0</td>
              </tr>
            ))}
          </tbody>
        </table>
      )}
    </>
  );
}
