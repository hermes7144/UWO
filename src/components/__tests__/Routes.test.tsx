import { render, screen, waitFor } from '@testing-library/react';
import { Route } from 'react-router-dom';
import { fakeRoutes } from '../../tests/fakeRoutes';
import { withAllContexts, withRouter } from '../../tests/utils';
import Routes from '../Routes';

describe('Routes', () => {
  const fakeUseRoute = jest.fn();

  afterEach(() => fakeUseRoute.mockReset());

  it('renders correctly', () => {
    fakeUseRoute.mockImplementation(() => {
      return {
        routesQuery: { data: fakeRoutes },
      };
    });
    const { asFragment } = renderRoutes();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders loading', () => {
    fakeUseRoute.mockImplementation(() => {
      return { routesQuery: { isLoading: true, data: fakeRoutes } };
    });
    renderRoutes();

    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });

  it('renders error', async () => {
    fakeUseRoute.mockImplementation(() => {
      return { routesQuery: { error: true } };
    });
    renderRoutes();

    await waitFor(() => {
      expect(screen.getByText('Something is wrong')).toBeInTheDocument();
    });
  });

  function renderRoutes() {
    return render(withAllContexts(withRouter(<Route path='/' element={<Routes country={'오스만'} region={'카리브해'} month={2} />} />), fakeUseRoute));
  }
});
