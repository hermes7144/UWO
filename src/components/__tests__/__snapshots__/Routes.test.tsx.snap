// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes renders correctly 1`] = `
<DocumentFragment>
  <table
    class="min-w-full text-center text-sm font-light"
  >
    <colgroup>
      <col
        width="5%"
      />
      <col
        width="5%"
      />
      <col
        width="5%"
      />
      <col
        width="50%"
      />
      <col
        width="10%"
      />
      <col
        width="10%"
      />
    </colgroup>
    <thead
      class="border-b bg-neutral-50 font-medium dark:border-neutral-500 dark:text-neutral-800"
    >
      <tr>
        <th
          class="px-1 py-4"
          scope="col"
        >
          국가
        </th>
        <th
          class="px-1 py-4"
          scope="col"
        >
          지역
        </th>
        <th
          class="px-1 py-4"
          scope="col"
        >
          기간
        </th>
        <th
          class="px-1 py-4"
          scope="col"
        >
          제목
        </th>
        <th
          class="px-1 py-4"
          scope="col"
        >
          작성일
        </th>
        <th
          class="px-1 py-4"
          scope="col"
        >
          추천
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="border-b dark:border-neutral-500"
      >
        <td
          class="whitespace-nowrap px-1 py-2"
        />
        <td
          class="whitespace-nowrap px-1 py-2"
        />
        <td
          class="whitespace-nowrap px-1 py-2"
        >
           ~ 월
        </td>
        <td
          class=" overflow-hidden px-2 py-4 text-left text-ellipsis whitespace-nowrap"
          style="max-width: 250px;"
        >
          <span
            class="cursor-pointer hover:underline"
          >
            title1
          </span>
        </td>
        <td
          class="whitespace-nowrap px-2 py-2"
        >
          23/03/27
        </td>
        <td
          class="whitespace-nowrap  px-2 py-2"
        >
          0
        </td>
      </tr>
      <tr
        class="border-b dark:border-neutral-500"
      >
        <td
          class="whitespace-nowrap px-1 py-2"
        />
        <td
          class="whitespace-nowrap px-1 py-2"
        />
        <td
          class="whitespace-nowrap px-1 py-2"
        >
           ~ 월
        </td>
        <td
          class=" overflow-hidden px-2 py-4 text-left text-ellipsis whitespace-nowrap"
          style="max-width: 250px;"
        >
          <span
            class="cursor-pointer hover:underline"
          >
            title2
          </span>
        </td>
        <td
          class="whitespace-nowrap px-2 py-2"
        >
          23/03/27
        </td>
        <td
          class="whitespace-nowrap  px-2 py-2"
        >
          0
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
